libsolidity/interface/CompilerStack.cpp

```diff
@@ -1300,11 +1300,34 @@ void CompilerStack::generateEVMFromIR(ContractDefinition const& _contract)
 
 	// TODO: support passing metadata
 	// TODO: use stack.assemble here!
-	yul::MachineAssemblyObject init;
-	yul::MachineAssemblyObject runtime;
-	std::tie(init, runtime) = stack.assembleAndGuessRuntime();
-	compiledContract.object = std::move(*init.bytecode);
-	compiledContract.runtimeObject = std::move(*runtime.bytecode);
+	compiledContract.evmAssembly = stack.assembleEVM();
+	solAssert(compiledContract.evmAssembly, "");
+	try
+	{
+		// Assemble deployment (incl. runtime)  object.
+		compiledContract.object = compiledContract.evmAssembly->assemble();
+	}
+	catch(evmasm::AssemblyException const&)
+	{
+		solAssert(false, "Assembly exception for bytecode");
+	}
+	solAssert(compiledContract.object.immutableReferences.empty(), "Leftover immutables.");
+
+	if (compiledContract.evmAssembly->numSubs() == 1)
```

Comment:
I think we should extend Assembly to store names of sub-assemblies and then access the runtime sub-assemlby by its name.